networkx/generators/directed.py

```diff
@@ -305,3 +308,83 @@ def _choose_node(G, distribution, delta, psum):
         
     return G
 
+
+def random_k_out_graph(n, k, alpha, self_loops=True):
```

Comment:
Did we ever decide on an API to make these functions have reproducible behavior?  #1550. It would be nice if we could at least provide the seed or the prng instance that the algorithm uses.